utils.h

```diff
@@ -25,6 +25,16 @@ typedef struct Crypto {
 	size_t (*decrypt)(void *param, const size_t size, void *dst, const void *src);
 } Crypto;
 
+struct Progress {
+	double elapsed_time;
+	size_t iterations_completed;
+	size_t iterations_total;
+	const char *lib_name;
+	double start_time;
+};
```

Comment:
To avoid the need to specify `struct` when declaring it:
```suggestion
typedef struct Progress {
	double elapsed_time;
	size_t iterations_completed;
	size_t iterations_total;
	const char *lib_name;
	double start_time;
} Progress;
```